elastic/kibana
# [Fleet] Fix bulk action dropdown

## Summary

Closes https://github.com/elastic/kibana/issues/164083
Related to https://github.com/elastic/kibana/issues/157844

This PR addresses two current issues affecting agent selection in Fleet UI:
1. When there are inactive agents that are not listed on the current page and "Select everything on all pages" is clicked, the count of actionable agents is incorrect (cf. [this comment](https://github.com/elastic/kibana/issues/164083#issuecomment-1711780591) for details). This can have two consequences:
   1. Incorrect and sometimes negative agent count in the "Actions" dropdown.
   2. Disabled menu items in the "Actions" dropdown.
2. The "Select everything on all pages button is incorrectly displayed when there are agents on managed policies on the page and there is no pagination (cf. [this comment](https://github.com/elastic/kibana/issues/164083#issuecomment-1711781808)).

## Testing steps

### Reproduce the bug (main branch)

1. Have a fleet server with a managed policy. One way of doing this is to create the usual fleet server policy with `fleet-server-policy` id and run the following in Dev Tools:
   ```
   PUT kbn:/api/fleet/agent_policies/fleet-server-policy
   {
     "name": "Managed agent policy",
     "description": "",
     "namespace": "default",
     "monitoring_enabled": [
       "logs",
       "metrics"
     ],
     "is_managed": true
   }
   ```
2. On another agent policy (`Agent policy 1`), enroll 30 agents, have them all inactive except for one. This can be achieved like so:
   - Enroll 30 agents with horde
   - Set agent policy inactivity timeout to 60 seconds
   - Kill horde agents
   - Enroll another agent (e.g. via a Docker container or a VM)
3. In Fleet's main page (agents list), show agents for all policies and hide inactive (default filters), so 2 agents are visible (fleet server + active agent).
4. Select all agents on first page:
   - Actions menu is enabled and correctly shows agent count as 1 (since fleet server is not selectable).
   - "Select everything on all pages" is displayed (incorrect 🐞).
5. Click "Select everything on all pages": the Actions menu is disabled and agent count is negative (-29) (incorrect 🐞).

### Test the fixes (this branch)

With the above setup:
1. In Fleet's main page (agents list), show agents for all policies and hide inactive (default filters), so 2 agents are visible (fleet server + active agent).
2. Select all agents on first page. Check that:
   - Actions menu is enabled and correctly shows agent count as 1 (since fleet server is not selectable).
   - "Select everything on all pages" is NOT displayed (fix ✅).
3. Enroll more agents with horde so that you have enough active agents to trigger pagination (you should still have your 30 inactive agents as well).
4. Select all agents on first page. The "Select everything on all pages" button should be displayed.
5. Click "Select everything on all pages": check that the Actions menu is enabled and shows the correct count of agents.

### Additional testing

The following scenarios should be checked as they depend on the area of the code that was amended:
- [Recently inactive agents popover](https://github.com/elastic/kibana/pull/149226)
- [Hosted agent count should not be included in agent count](https://github.com/elastic/kibana/issues/152180)

### Checklist
